_inc/client/state/connection/actions.js

```diff
@@ -79,12 +78,13 @@ export const disconnectSite = () => {
 			type: DISCONNECT_SITE
 		} );
 		dispatch( createNotice( 'is-info', __( 'Disconnecting Jetpack' ), { id: 'disconnect-jetpack' } ) );
-		return restApi.disconnectSite().then( disconnectingSite => {
-			dispatch( {
-				type: DISCONNECT_SITE_SUCCESS,
-				disconnectingSite: disconnectingSite
-			} );
-			dispatch( removeNotice( 'disconnect-jetpack' ) );
+		return restApi.disconnectSite().then( () => {
+
+			// Because the sidebar navigation is not connected to Jetpack state tree
+			// it is easiest to just reload the page on disconnection to remove all
+			// unneeded navigation items, like Settings.
+			window.scrollTo( 0, 0 );
```

Comment:
These two lines (`window.something`), could be moved to a Component, reacting to a props change, as we talked on Slack regarding where to put jQuery calls. 
Maybe in `Main` or `Masthead`'s `componentDidUpdate` and checking connection status in  old props vs current props